install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```
@@ -57,3 +57,13 @@ spec:
             severity: warning
           annotations:
             message: "System memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90% of the reservation. Reserved memory ensures system processes can function even when the node is fully allocated and protects against workload out of memory events impacting the proper functioning of the node. The default reservation is expected to be sufficient for most configurations and should be increased (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-managing.html) when running nodes with high numbers of pods (either due to rate of change or at steady state)."
+    - name: containers-oom-killed
+      rules:
+        - alert: ContainersOOMKilled
+          expr: |
+            sum by (node) (container_runtime_crio_containers_oom_total) > 5
+          for: 15m
+          labels:
+            severity: warning
+          annotations:
+            message: "{{ $value | humanize }} containers have been killed on {{ $labels.node }} because they ran out of memory (OOM). Please make sure to allocate adequate enough memory for the pods or try to reduce the total number of pods."
```

Yes, let's add an information that overcommitting the node is not recommended:

```suggestion
            message: "{{ $value | humanize }} containers have been killed on {{ $labels.node }} because they ran out of memory (OOM). Please make sure to limit the memory for the containers to not overcommit the node or try to reduce the total number of pods per node."
```